Sources/Plasma/FeatureLib/pfPython/cyMiscGlue4.cpp

```diff
@@ -402,37 +402,71 @@ PYTHON_GLOBAL_METHOD_DEFINITION(PtDebugAssert, args, "Params: cond, msg\nDebug o
     PYTHON_RETURN_NONE;
 }
 
-PYTHON_GLOBAL_METHOD_DEFINITION_WKEY(PtDebugPrint, args, kwargs, "Params: *msgs, **kwargs\nPrints msgs to the Python log given the message's level")
+PYTHON_GLOBAL_METHOD_DEFINITION_WKEY(PtDebugPrint, args, kwargs, "Params: *msgs, level, sep, end\n"
+                                     "Prints msgs to the Python log given the message's level, "
+                                     "optionally separated and terminated by the given strings")
 {
     uint32_t  level = cyMisc::kErrorLevel;
+    ST::string sep = ST_LITERAL(" ");
+    ST::string end = ST_LITERAL("\n");
 
     do {
-        // Grabbin' levelz
         if (kwargs && PyDict_Check(kwargs)) {
-            PyObject* value = PyDict_GetItem(kwargs, PyString_FromString("level"));
+            PyObject* value = PyDict_GetItemString(kwargs, "level");
             if (value) {
                 if (PyInt_Check(value))
                     level = PyInt_AsLong(value);
                 else
                     break;
             }
+
+            value = PyDict_GetItemString(kwargs, "sep");
+            if (value) {
+                value = PyObject_Str(value);
```

Comment:
Nice addition, but if you want to be strict about compatibility with `print()`, Python3 does actually require that `sep` and `end` are strings (it even rejects bytes)

Reply:
I didn't realize that. Do you have a preference either way?

Reply:
I'd probably prefer to keep it like `print()`, to make future changes easier